returnn/config.py

```diff
@@ -328,7 +344,6 @@ def check_consistency(self):
             assert key not in self.post_config, (
                 "%s in post_config would overwrite existing entry in config" % key
             )
-        assert not (self.staged_network_dict and "network" in self.config)
```

Comment:
>  In this case we neither have "network" in the dict nor set staged_network_dict, so we have to remove the assert.

this assert should only break if both `staged_network_dict` and `network` are given. Giving neither should be fine as 
```
assert not (False and False) => assert True
```

Reply:
I dont see a reason both being not given to ever make sense so I agree, this should be changed. 
Does python have built in XOR for these types? Otherwise I guess we should go with
`assert (self.staged_network_dict and not "network" in self.config) or (not self.staged_network_dict and "network" in self.config)`

Whats possible (but maybe a bit too hidden) is: `assert (bool(self.staged_network_dict) ^ bool("network" in self.config))`

Reply:
But this is exactly what I wrote in the comment, there is a case now where both are not given. Not sure why this assert was removed in my code, Willi is right I can just add it again and everything should be fine.